libraries/AP_HAL/I2CDevice.h

```diff
@@ -27,6 +27,8 @@ namespace AP_HAL {
 
 class I2CDevice : public Device {
 public:
+    I2CDevice() : Device(I2C) { }
```

Comment:
Any particular reason to not use dynamic_cast instead of to add an enum in the base class? What if device is neither spi, i2c in the future? Will have to keep growing the enum.